src/rpdk/core/contract/suite/handler_update.py

```diff
@@ -45,7 +45,8 @@ def contract_update_read_success(updated_resource, resource_client):
     _create_request, _created_model, _update_request, updated_model = updated_resource
     assert resource_client.is_primary_identifier_equal(
         resource_client.primary_identifier_paths, _created_model, updated_model
-    )
+    ), "The primaryIdentifier returned in every progress event must match\
```

Comment:
this is not being checked for every progress evert but only when a handler returns failed/success. We might ant to rephrase this to reflect that

Reply:
oh, good catch